### PR TITLE
feat(sbt): support bumpVersion

### PR DESCRIPTION
**Pull request by marius-carp**
## Changes:

Support `bumpVersion` in sbt managers.

## Context:

Described here: https://github.com/renovatebot/renovate/issues/7962#issuecomment-745998635

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by marius-carp** [comment]
@rarkins any news on this one, is there something else that I need to cover?

File: lib/manager/sbt/update.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    'Checking if we should bump build.sbt version'
+  );
+  let newVersion: string;
+  try {
+    logger.info({
+      content,
+      currentValue,
+      bumpVersion,
+    });
+
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:23]:
```suggestion
```

File: lib/manager/sbt/update.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    newVersion = inc(currentValue, bumpVersion as ReleaseType);
+
+    if (!newVersion) {
+      throw new Error('Version incremental failed');
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:27]: log a warning and return, no error necessary

File: lib/manager/sbt/update.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    logger.info({ newVersion });
+
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:31]:
```suggestion
```

File: lib/manager/sbt/update.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const bumpedContent = content.replace(
+      /^(version\s*:=\s*).*$/m,
+      `$1"${newVersion}"`
+    );
+
+    if (bumpedContent === content) {
+      logger.info('Version was already bumped');
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:38]:
```suggestion
      logger.debug('Version was already bumped');
```

File: lib/manager/sbt/update.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    } else {
+      logger.info('Bumped build.sbt version');
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:40]:
```suggestion
      logger.debug({ newVersion }, 'Bumped build.sbt version');
```

File: lib/manager/sbt/update.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    return bumpedContent;
+  } catch (err) {
+    logger.warn(
+      {
+        content,
+        currentValue,
+        bumpVersion,
+      },
+      'Failed to bumpVersion'
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:51]:
```suggestion
      'Failed to sbt.bumpVersion'
```

File: lib/manager/sbt/update.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    );
+
+    return content;
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:54]:
```suggestion
    return { bumpedContent: content };
```

File: lib/manager/sbt/update.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+      `$1"${newVersion}"`
+    );
+
+    if (bumpedContent === content) {
+      logger.info('Version was already bumped');
+    } else {
+      logger.info('Bumped build.sbt version');
+    }
+
+    return bumpedContent;
```
Comment (rarkins) [inline lib/manager/sbt/update.ts:43]:
```suggestion
    return { bumpedContent };
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/sbt/update.ts (hunk 2 above)
Comment (marius-carp) [reply lib/manager/sbt/update.ts]: this means that the `try-catch` statement is not needed anymore

File: lib/manager/sbt/update.ts (hunk 1 above)
Comment (marius-carp) [reply lib/manager/sbt/update.ts]: :+1:

File: lib/manager/sbt/update.ts (hunk 3 above)
Comment (marius-carp) [reply lib/manager/sbt/update.ts]: :+1:

File: lib/manager/sbt/update.ts (hunk 4 above)
Comment (marius-carp) [reply lib/manager/sbt/update.ts]: :+1:

File: lib/manager/sbt/update.ts (hunk 5 above)
Comment (marius-carp) [reply lib/manager/sbt/update.ts]: :+1:

**Comment by marius-carp** [comment]
@rarkins I addressed all the comments, please have a second look. Thank you!

**Pull request merged**